### PR TITLE
Introduction of repo-tool packages with `api-report`

### PR DESCRIPTION
**Pull request by Sarabadu**
## Hey, I just made a Pull Request!

Replace the `scripts/api-extractor.ts` with the new CLI package `backstage-repo-tools api-reports` command.

Today the `api-report` tooling runs directly from a script. converting it to a package will make it easy to share tooling in case of needed and give better dependency management 

TODO:
the script was moved "as is" and will require future work to add tests and refactor on more readable parts

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by freben — COMMENTED** [review]
Nice!

File: package.json [hunk 1]
```diff
@@ -10,8 +10,9 @@
     "start-backend": "yarn workspace example-backend start",
     "build:backend": "yarn workspace backend build",
     "build:all": "backstage-cli repo build --all",
+    "build": "backstage-cli repo build --all",
```
[the remark this answers is not in the document]
Comment (Sarabadu) [reply package.json:13]: yes a rebase error I'm looking at it, thanks!!

File: packages/repo-tools/package.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  },
+  "bin": {
+    "backstage-repo-tools": "bin/backstage-repo-tools"
+  },
+  "dependencies": {
+    "@backstage/errors": "workspace:^",
+    "@microsoft/api-documenter": "^7.17.11",
+    "@microsoft/api-extractor": "^7.23.0",
+    "@microsoft/api-extractor-model": "^7.17.2",
+    "@microsoft/tsdoc": "0.14.1",
```
[the remark this answers is not in the document]
Comment (Sarabadu) [reply packages/repo-tools/package.json:37]: Version `0.14.2` brings a lot of issues on typing, will look at it in detail

**Comment by freben** [comment]
You may want to turn on autoformatting on save in your editor (see warnings about `prettier`)

File: packages/repo-tools/package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sarabadu) [reply packages/repo-tools/package.json:37]: the
```
    "@microsoft/api-documenter": "^7.17.11",
    "@microsoft/api-extractor": "^7.23.0",
    "@microsoft/api-extractor-model": "^7.17.2",
```

has the fixed version `0.14.1` seems like scripts were not included in the TSC compilation before

**Review by djamaile — APPROVED** [review]
You work fast ^ ^. Tested it locally and it seems to be working for me! I was wondering that maybe we still want to create a sep branch that on purpose has an error with the API reports and see if it fails as expected? Not sure what you think 😄

File: packages/repo-tools/src/commands/api-reports/api-extractor.ts [hunk 3]
```diff
@@ -210,7 +223,7 @@ const ALLOW_WARNINGS = [
 async function resolvePackagePath(
   packagePath: string,
 ): Promise<string | undefined> {
-  const projectRoot = resolvePath(__dirname, '..');
+  const projectRoot = resolvePath(process.cwd());
```
Comment (Rugvip) [inline packages/repo-tools/src/commands/api-reports/api-extractor.ts:226]: Could make this a followup, but I think it's best if we find the true project root using [`findPaths`](https://github.com/backstage/backstage/blob/5bb2d0f32b4b6522f17b76283bc58f2213ad6726/packages/cli-common/src/paths.ts#L118) from `cli-common`. That way you can run this from anywhere in the repo.

**Review by Rugvip — APPROVED** [review]
Awesome stuff, thank you! 😁

File: packages/repo-tools/src/commands/api-reports/api-extractor.ts (hunk 3 above)
Comment (Sarabadu) [reply packages/repo-tools/src/commands/api-reports/api-extractor.ts:226]: Yes, thanks I will include those on follow-up refactor

**Pull request merged**